lib/ios/BackpackReactNative/Classes/CalendarBridge/RCTBPKCalendar.m

```diff
@@ -160,6 +139,73 @@ - (void)setRct_disabledDates:(RCTBPKDateMatcher *_Nullable)rct_disabledDates {
 
+        self.nativeUpdateRequired = YES;
+    }
+}
+
+-(void)setNativeSelectedDates:(NSArray<NSDate *> *)selectedDates {
+    NSMutableArray *simpleDates = [[NSMutableArray alloc] initWithCapacity:selectedDates.count];
+
+    for (NSDate *utcDate in selectedDates) {
+        NSDate *localDate = [RCTBPKCalendarDateUtils convertDateToLocal:utcDate
+                                                          localCalendar:self.gregorian
+                                                            utcCalendar:self.utcCalendar];
+        BPKSimpleDate *simpleDate = [[BPKSimpleDate alloc] initWithDate:localDate forCalendar:self.gregorian];
+        [simpleDates addObject:simpleDate];
+    }
+
+    [super setSelectedDates:simpleDates];
+}
+
+- (void)setNativeMinDate:(nullable NSDate *)minDate {
+    // `BPKCalendar` requires `minDate` to be nonnull
+    // but the RN interface supports `null` for the `minDate`
+    // prop thus we have to ensure we have an explicit default
+    // here. This default is aligned with the Android implementation.
+    if (minDate == nil) {
+        NSDate *today = [[NSDate alloc] init];
+        BPKSimpleDate *date = [[BPKSimpleDate alloc] initWithDate:today forCalendar:self.gregorian];
+
+        [super setMinDate:date];
+        return;
+    }
+
+    NSDate *localDate = [RCTBPKCalendarDateUtils convertDateToLocal:minDate
+                                                      localCalendar:self.gregorian
+                                                        utcCalendar:self.utcCalendar];
+    BPKSimpleDate *simpleDate = [[BPKSimpleDate alloc] initWithDate:localDate forCalendar:self.gregorian];
+    [super setMinDate:simpleDate];
+}
+
+- (void)setNativeMaxDate:(nullable NSDate *)maxDate {
+    // `BPKCalendar` requires `maxDate` to be nonnull
+    // but the RN interface supports `null` for the `maxDate`
+    // prop thus we have to ensure we have an explicit default
+    // here. This default is aligned with the Android implementation.
+    if (maxDate == nil) {
+        NSDate *today = [[NSDate alloc] init];
+        NSDate *oneYearFromNow = [self.gregorian dateByAddingUnit:NSCalendarUnitYear value:1 toDate:today options:0];
+        BPKSimpleDate *date = [[BPKSimpleDate alloc] initWithDate:oneYearFromNow forCalendar:self.gregorian];
+
+        [super setMaxDate:date];
+        return;
+    }
+
+    NSDate *localDate = [RCTBPKCalendarDateUtils convertDateToLocal:maxDate
+                                                      localCalendar:self.gregorian
+                                                        utcCalendar:self.utcCalendar];
+    BPKSimpleDate *simpleDate = [[BPKSimpleDate alloc] initWithDate:localDate forCalendar:self.gregorian];
+    [super setMaxDate:simpleDate];
+}
+
+- (void)didSetProps:(NSArray<NSString *> *)changedProps {
+    if (self.nativeUpdateRequired) {
+        self.nativeUpdateRequired = false;
+
+        [super setSelectionType:self.rct_selectionType];
```

Comment:
I don't think `set...` is the best name for these because it sounds like a setter for a class property. Maybe we should rename these `update...`?

Reply:
I don't really agree, this seems reasonable but we could just write it as `super.selectionType = self.rct_selectionType`

Reply:
Oh I put the comment on the wrong line! Yeah this one is fine, but I meant the ones below. `setNativeMinDate`, `setNativeMaxDate`, `setNativeSelectedDates`

Reply:
Yeah I guess we can do that

Reply:
Done